src/fpm/manifest/profiles.f90

```diff
@@ -0,0 +1,732 @@
+                call syntax_error(error, "files has to be a table")
+                return
+              end if
+              call files%get_keys(file_list)
+              allocate(file_scope_flags(size(file_list)))
+              do ifile=1,size(file_list)
+                file_name = file_list(ifile)%key
+                call get_value(files, file_name, file_flags, stat=stat)
+                if (stat /= toml_stat%success) then
+                  call syntax_error(error, "file scope flags has to be a key-value pair")
+                  return
+                end if
+                associate(cur_file=>file_scope_flags(ifile))
+                  if (.not.(path.eq."")) file_name = join_path(path, file_name)
+                  cur_file%file_name = file_name
+                  cur_file%flags = file_flags
+                end associate
+              end do
+            end if
+          end do
+        end if
+
+        if (.not.allocated(flags)) flags=''
+        if (.not.allocated(c_flags)) c_flags=''
+        if (.not.allocated(link_time_flags)) link_time_flags=''
+
+        if (allocated(file_scope_flags)) then
+          profiles(profindex) = new_profile(profile_name, compiler_name, os_type, &
+                 & flags, c_flags, link_time_flags, file_scope_flags)
+        else
+          profiles(profindex) = new_profile(profile_name, compiler_name, os_type, flags, c_flags, link_time_flags)
+        end if
+        profindex = profindex + 1
+      end subroutine get_flags
+
+      !> Traverse operating system tables
+      subroutine traverse_oss(profile_name, compiler_name, os_list, table, error, profiles_size, profiles, profindex)
+        
+        !> Name of profile
+        character(len=:), allocatable, intent(in) :: profile_name
+
+        !> Name of compiler
+        character(len=:), allocatable, intent(in) :: compiler_name
+
+        !> List of OSs in table with profile name and compiler name given
+        type(toml_key), allocatable, intent(in) :: os_list(:)
+
+        !> Table containing OS tables
+        type(toml_table), pointer, intent(in) :: table
+
+        !> Error handling
+        type(error_t), allocatable, intent(out) :: error
+
+        !> Number of profiles in list of profiles
+        integer, intent(inout), optional :: profiles_size
+
+        !> List of profiles
+        type(profile_config_t), allocatable, intent(inout), optional :: profiles(:)
+
+        !> Index in the list of profiles
+        integer, intent(inout), optional :: profindex
+        
+        type(toml_key), allocatable :: key_list(:)
+        character(len=:), allocatable :: os_name
+        type(toml_table), pointer :: os_node
+        character(len=:), allocatable :: flags
+        integer :: ios, stat, os_type
+        logical :: is_valid, key_val_added, is_key_val
+
+        if (size(os_list)<1) return
+        key_val_added = .false.
+        do ios = 1, size(os_list)
+          os_name = lower(os_list(ios)%key)
```

Comment:
Is the OS name really case insensitive? What Happens if there are `Linux`, `LINUX` and `linux` in the same profile?

Reply:
Then all three profiles are created, but when the `find_profile` subroutine is called, the last one of them (therefore the one with `linux`) is taken.